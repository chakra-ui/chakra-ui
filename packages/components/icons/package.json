{
  "name": "@chakra-ui/icons",
  "version": "2.0.17",
  "description": "Chakra UI icons",
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "homepage": "https://github.com/chakra-ui/chakra-ui#readme",
  "license": "MIT",
  "main": "src/index.ts",
  "sideEffects": false,
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/chakra-ui/chakra-ui.git",
    "directory": "packages/components/icons"
  },
  "bugs": {
    "url": "https://github.com/chakra-ui/chakra-ui/issues"
  },
  "peerDependencies": {
    "@chakra-ui/system": ">=2.0.0",
    "react": ">=18"
  },
  "devDependencies": {
    "@types/react": "^18.0.1",
    "@chakra-ui/system": "workspace:*",
    "@chakra-ui/layout": "workspace:*",
    "react": "^18.0.0",
    "clean-package": "2.1.1"
  },
  "dependencies": {
    "@chakra-ui/icon": "workspace:*"
  },
  "scripts": {
    "build": "tsup src --dts",
    "dev": "pnpm build:fast -- --watch",
    "clean": "rimraf dist .turbo",
    "typecheck": "tsc --noEmit",
    "build:fast": "tsup src",
    "prepack": "clean-package",
    "postpack": "clean-package restore",
    "typedocs": "tsx ../../../scripts/generate-type-docs.ts"
  },
  "clean-package": "../../../clean-package.config.json",
  "tsup": {
    "clean": true,
    "target": "es2019",
    "format": [
      "cjs",
      "esm"
    ]
  }
}
