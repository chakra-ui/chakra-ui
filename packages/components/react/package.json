{
  "name": "@chakra-ui/react",
  "version": "2.4.9",
  "description": "Responsive and accessible React UI components built with React and Emotion",
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@chakra-ui/accordion": "workspace:*",
    "@chakra-ui/alert": "workspace:*",
    "@chakra-ui/avatar": "workspace:*",
    "@chakra-ui/breadcrumb": "workspace:*",
    "@chakra-ui/button": "workspace:*",
    "@chakra-ui/card": "workspace:*",
    "@chakra-ui/checkbox": "workspace:*",
    "@chakra-ui/close-button": "workspace:*",
    "@chakra-ui/control-box": "workspace:*",
    "@chakra-ui/counter": "workspace:*",
    "@chakra-ui/css-reset": "workspace:*",
    "@chakra-ui/editable": "workspace:*",
    "@chakra-ui/form-control": "workspace:*",
    "@chakra-ui/hooks": "workspace:*",
    "@chakra-ui/icon": "workspace:*",
    "@chakra-ui/image": "workspace:*",
    "@chakra-ui/input": "workspace:*",
    "@chakra-ui/layout": "workspace:*",
    "@chakra-ui/live-region": "workspace:*",
    "@chakra-ui/media-query": "workspace:*",
    "@chakra-ui/menu": "workspace:*",
    "@chakra-ui/modal": "workspace:*",
    "@chakra-ui/number-input": "workspace:*",
    "@chakra-ui/pin-input": "workspace:*",
    "@chakra-ui/popover": "workspace:*",
    "@chakra-ui/popper": "workspace:*",
    "@chakra-ui/portal": "workspace:*",
    "@chakra-ui/progress": "workspace:*",
    "@chakra-ui/provider": "workspace:*",
    "@chakra-ui/radio": "workspace:*",
    "@chakra-ui/react-env": "workspace:*",
    "@chakra-ui/select": "workspace:*",
    "@chakra-ui/skeleton": "workspace:*",
    "@chakra-ui/slider": "workspace:*",
    "@chakra-ui/spinner": "workspace:*",
    "@chakra-ui/stat": "workspace:*",
    "@chakra-ui/styled-system": "workspace:*",
    "@chakra-ui/switch": "workspace:*",
    "@chakra-ui/system": "workspace:*",
    "@chakra-ui/table": "workspace:*",
    "@chakra-ui/tabs": "workspace:*",
    "@chakra-ui/tag": "workspace:*",
    "@chakra-ui/textarea": "workspace:*",
    "@chakra-ui/theme": "workspace:*",
    "@chakra-ui/theme-utils": "workspace:*",
    "@chakra-ui/toast": "workspace:*",
    "@chakra-ui/tooltip": "workspace:*",
    "@chakra-ui/transition": "workspace:*",
    "@chakra-ui/utils": "workspace:*",
    "@chakra-ui/visually-hidden": "workspace:*"
  },
  "peerDependencies": {
    "@emotion/react": "^11.0.0",
    "@emotion/styled": "^11.0.0",
    "framer-motion": ">=4.0.0",
    "react": ">=18",
    "react-dom": ">=18"
  },
  "author": "Segun Adebayo <sage@adebayosegun.com>",
  "license": "MIT",
  "homepage": "https://chakra-ui.com/",
  "repository": {
    "type": "git",
    "url": "https://github.com/chakra-ui/chakra-ui",
    "directory": "packages/components/react"
  },
  "keywords": [
    "react",
    "ui",
    "design-system",
    "react-components",
    "uikit",
    "accessible",
    "components",
    "emotion",
    "library",
    "design-system"
  ],
  "storybook": {
    "title": "Chakra UI",
    "url": "https://storybook.chakra-ui.com"
  },
  "devDependencies": {
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "framer-motion": "^6.2.9",
    "react": "^18.0.0",
    "react-dom": "^18.0.0",
    "clean-package": "2.1.1"
  },
  "scripts": {
    "build": "tsup src --dts",
    "dev": "pnpm build:fast -- --watch",
    "clean": "rimraf dist .turbo",
    "typecheck": "tsc --noEmit",
    "build:fast": "tsup src",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "clean-package": "../../../clean-package.config.json",
  "tsup": {
    "clean": true,
    "target": "es2019",
    "format": [
      "cjs",
      "esm"
    ]
  }
}
