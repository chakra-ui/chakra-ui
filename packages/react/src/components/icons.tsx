"use client"

import { type HTMLChakraProps, chakra } from "../styled-system"
import { Icon } from "./icon/icon"

/**
   
 This file is internal and should not be exported to the consumer.
 
 We only add icons really needed for the component to looks good by default.
 We're not in the business of creating icons.
 
 Please find ideas from https://react-icons.github.io/react-icons/ 

 */

interface SvgProps extends HTMLChakraProps<"svg"> {}

export const CheckIcon = (props: SvgProps) => {
  return (
    <Icon
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      {...props}
    >
      <path d="M20 6 9 17l-5-5" />
    </Icon>
  )
}

export const ChevronUpIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="2"
    strokeLinecap="round"
    strokeLinejoin="round"
    {...props}
  >
    <path d="m18 15-6-6-6 6" />
  </Icon>
)

export const ChevronDownIcon = (props: SvgProps) => {
  return (
    <Icon
      width="1em"
      height="1em"
      viewBox="0 0 24 24"
      fill="none"
      stroke="currentColor"
      strokeWidth="2"
      strokeLinecap="round"
      strokeLinejoin="round"
      {...props}
    >
      <path d="m6 9 6 6 6-6" />
    </Icon>
  )
}

export const ChevronRightIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="2"
    strokeLinecap="round"
    strokeLinejoin="round"
    {...props}
  >
    <path d="m9 18 6-6-6-6" />
  </Icon>
)

export const EllpsisIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="2"
    strokeLinecap="round"
    strokeLinejoin="round"
    {...props}
  >
    <circle cx="12" cy="12" r="1" />
    <circle cx="19" cy="12" r="1" />
    <circle cx="5" cy="12" r="1" />
  </Icon>
)

export const ArrowUpIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="3"
    strokeLinecap="round"
    strokeLinejoin="round"
    {...props}
  >
    <path d="m5 12 7-7 7 7"></path>
    <path d="M12 19V5"></path>
  </Icon>
)

export const ArrowDownIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="none"
    stroke="currentColor"
    strokeWidth="3"
    strokeLinecap="round"
    strokeLinejoin="round"
    {...props}
  >
    <path d="M12 5v14"></path>
    <path d="m19 12-7 7-7-7"></path>
  </Icon>
)

export const CheckCircleIcon = (props: SvgProps) => (
  <chakra.svg
    stroke="currentColor"
    fill="currentColor"
    strokeWidth="0"
    viewBox="0 0 24 24"
    height="1em"
    width="1em"
    {...props}
  >
    <path d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11.0026 16L6.75999 11.7574L8.17421 10.3431L11.0026 13.1716L16.6595 7.51472L18.0737 8.92893L11.0026 16Z" />
  </chakra.svg>
)

export const WarningIcon = (props: SvgProps) => (
  <chakra.svg
    stroke="currentColor"
    fill="currentColor"
    strokeWidth="0"
    viewBox="0 0 24 24"
    height="1em"
    width="1em"
    {...props}
  >
    <path d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11 15H13V17H11V15ZM11 7H13V13H11V7Z" />
  </chakra.svg>
)

export const InfoIcon = (props: SvgProps) => (
  <chakra.svg
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="currentColor"
    stroke="currentColor"
    strokeWidth="0"
    {...props}
  >
    <path d="M12 22C6.47715 22 2 17.5228 2 12C2 6.47715 6.47715 2 12 2C17.5228 2 22 6.47715 22 12C22 17.5228 17.5228 22 12 22ZM12 20C16.4183 20 20 16.4183 20 12C20 7.58172 16.4183 4 12 4C7.58172 4 4 7.58172 4 12C4 16.4183 7.58172 20 12 20ZM11 7H13V9H11V7ZM11 11H13V17H11V11Z" />
  </chakra.svg>
)

export const QuoteIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="currentColor"
    {...props}
  >
    <path d="M6.5 10c-.223 0-.437.034-.65.065.069-.232.14-.468.254-.68.114-.308.292-.575.469-.844.148-.291.409-.488.601-.737.201-.242.475-.403.692-.604.213-.21.492-.315.714-.463.232-.133.434-.28.65-.35l.539-.222.474-.197-.485-1.938-.597.144c-.191.048-.424.104-.689.171-.271.05-.56.187-.882.312-.318.142-.686.238-1.028.466-.344.218-.741.4-1.091.692-.339.301-.748.562-1.05.945-.33.358-.656.734-.909 1.162-.293.408-.492.856-.702 1.299-.19.443-.343.896-.468 1.336-.237.882-.343 1.72-.384 2.437-.034.718-.014 1.315.028 1.747.015.204.043.402.063.539l.025.168.026-.006A4.5 4.5 0 1 0 6.5 10zm11 0c-.223 0-.437.034-.65.065.069-.232.14-.468.254-.68.114-.308.292-.575.469-.844.148-.291.409-.488.601-.737.201-.242.475-.403.692-.604.213-.21.492-.315.714-.463.232-.133.434-.28.65-.35l.539-.222.474-.197-.485-1.938-.597.144c-.191.048-.424.104-.689.171-.271.05-.56.187-.882.312-.317.143-.686.238-1.028.467-.344.218-.741.4-1.091.692-.339.301-.748.562-1.05.944-.33.358-.656.734-.909 1.162-.293.408-.492.856-.702 1.299-.19.443-.343.896-.468 1.336-.237.882-.343 1.72-.384 2.437-.034.718-.014 1.315.028 1.747.015.204.043.402.063.539l.025.168.026-.006A4.5 4.5 0 1 0 17.5 10z" />
  </Icon>
)

export const StarIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="currentColor"
    {...props}
  >
    <path d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
  </Icon>
)

export const ErrorIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="currentColor"
    {...props}
  >
    <path d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z" />
  </Icon>
)

export const CloseIcon = (props: SvgProps) => (
  <Icon
    width="1em"
    height="1em"
    viewBox="0 0 24 24"
    fill="currentColor"
    {...props}
  >
    <path
      fillRule="evenodd"
      clipRule="evenodd"
      d="M18.7071 6.70711C19.0976 6.31658 19.0976 5.68342 18.7071 5.29289C18.3166 4.90237 17.6834 4.90237 17.2929 5.29289L12 10.5858L6.70711 5.29289C6.31658 4.90237 5.68342 4.90237 5.29289 5.29289C4.90237 5.68342 4.90237 6.31658 5.29289 6.70711L10.5858 12L5.29289 17.2929C4.90237 17.6834 4.90237 18.3166 5.29289 18.7071C5.68342 19.0976 6.31658 19.0976 6.70711 18.7071L12 13.4142L17.2929 18.7071C17.6834 19.0976 18.3166 19.0976 18.7071 18.7071C19.0976 18.3166 19.0976 17.6834 18.7071 17.2929L13.4142 12L18.7071 6.70711Z"
    />
  </Icon>
)
