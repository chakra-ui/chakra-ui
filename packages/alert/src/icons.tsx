import * as React from "react"
import { Icon, IconProps } from "@chakra-ui/icon"

export const CheckIcon: React.FC<IconProps> = (props) => (
  <Icon viewBox="0 0 24 24" {...props}>
    <path
      fill="currentColor"
      d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
    />
  </Icon>
)

export const InfoIcon: React.FC<IconProps> = (props) => (
  <Icon viewBox="0 0 24 24" {...props}>
    <path
      fill="currentColor"
      d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
    />
  </Icon>
)

export const WarningIcon: React.FC<IconProps> = (props) => (
  <Icon viewBox="0 0 24 24" {...props}>
    <path
      fill="currentColor"
      d="M11.983,0a12.206,12.206,0,0,0-8.51,3.653A11.8,11.8,0,0,0,0,12.207,11.779,11.779,0,0,0,11.8,24h.214A12.111,12.111,0,0,0,24,11.791h0A11.766,11.766,0,0,0,11.983,0ZM10.5,16.542a1.476,1.476,0,0,1,1.449-1.53h.027a1.527,1.527,0,0,1,1.523,1.47,1.475,1.475,0,0,1-1.449,1.53h-.027A1.529,1.529,0,0,1,10.5,16.542ZM11,12.5v-6a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Z"
    />
  </Icon>
)
