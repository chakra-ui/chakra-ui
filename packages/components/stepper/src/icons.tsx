import { ComponentProps } from "react"

export function CheckIcon(props: ComponentProps<"svg">) {
  return (
    <svg
      stroke="currentColor"
      fill="currentColor"
      strokeWidth="0"
      viewBox="0 0 20 20"
      aria-hidden="true"
      height="1em"
      width="1em"
      {...props}
    >
      <path
        fillRule="evenodd"
        d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
        clipRule="evenodd"
      />
    </svg>
  )
}
