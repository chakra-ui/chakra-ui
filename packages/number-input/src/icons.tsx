import * as React from "react"
import { Icon, IconProps } from "@chakra-ui/icon"

export const TriangleDownIcon: React.FC<IconProps> = (props) => (
  <Icon viewBox="0 0 24 24" {...props}>
    <path
      fill="currentColor"
      d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
    />
  </Icon>
)

export const TriangleUpIcon: React.FC<IconProps> = (props) => (
  <Icon viewBox="0 0 24 24" {...props}>
    <path
      fill="currentColor"
      d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
    />
  </Icon>
)
