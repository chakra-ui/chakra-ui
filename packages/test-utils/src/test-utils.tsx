import theme from "@chakra-ui/preset-base"
import { GlobalStyle, ThemeProvider } from "@chakra-ui/system"
import CSSReset from "@chakra-ui/css-reset"
import "@testing-library/jest-dom/extend-expect"
import { render, RenderOptions } from "@testing-library/react"
import * as React from "react"
import { toHaveNoViolations } from "jest-axe"

expect.extend(toHaveNoViolations)

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <CSSReset />
    <GlobalStyle />
    {children}
  </ThemeProvider>
)

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options })

export * from "@testing-library/react"

export {act as invoke, renderHook} from "@testing-library/react-hooks"
export type  { RenderHookOptions, RenderHookResult} from "@testing-library/react-hooks"

export { default as userEvent } from "@testing-library/user-event"

export { customRender as render }

export * from "jest-axe"
