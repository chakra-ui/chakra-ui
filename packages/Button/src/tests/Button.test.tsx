import * as React from "react"
import { render } from "@chakra-ui/test-utils"
import { EmailIcon, ArrowForwardIcon } from "@chakra-ui/icons"
import { Button, ButtonGroup } from ".."

test("Button renders correctly", () => {
  const { asFragment } = render(<Button />)
  expect(asFragment()).toMatchSnapshot()
})

test("Button renders with icons", () => {
  const { asFragment } = render(
    <ButtonGroup>
      <Button leftIcon={EmailIcon}>Email</Button>
      <Button rightIcon={ArrowForwardIcon}>Arrow Forward</Button>
    </ButtonGroup>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("shows spinner if isLoading", () => {
  const { asFragment, getByText } = render(<Button isLoading>Email</Button>)
  expect(asFragment()).toMatchSnapshot()

  // children text is hidden
  expect(getByText("Email")).not.toBeVisible()

  // "Loading..." visually hidden label shown
  getByText("Loading...")
})

test("shows spinner and loading text if isLoading and loadingText", () => {
  const { asFragment, queryByText, getByText } = render(
    <Button isLoading loadingText="Submitting">
      Submit
    </Button>,
  )
  expect(asFragment()).toMatchSnapshot()

  // children text is replaced by `loadingText`
  getByText("Submitting")
  expect(queryByText("Submit")).toBeNull()
})

test("has the proper aria attributes", () => {
  const { rerender, getByRole } = render(<Button>Hello</Button>)

  // button has role="button"
  const button = getByRole("button")
  expect(button).not.toHaveAttribute("aria-disabled")

  // isLoading sets aria-disabled="true"
  rerender(<Button isLoading>Hello</Button>)
  expect(button).toHaveAttribute("aria-disabled", "true")

  // isDisabled sets aria-disabled="true"
  rerender(<Button isDisabled>Hello</Button>)
  expect(button).toHaveAttribute("aria-disabled", "true")
})
