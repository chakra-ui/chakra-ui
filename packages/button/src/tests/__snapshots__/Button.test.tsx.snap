// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="css-1ax3zce"
    type="button"
  />
</DocumentFragment>
`;

exports[`Button renders with icons 1`] = `
<DocumentFragment>
  <div
    class="css-1baulvz"
  >
    <button
      class="css-w134wy"
      type="button"
    >
      <svg
        class="css-4e9yzq"
        data-chakra-custom-icon=""
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <g
          fill="currentColor"
        >
          <path
            d="M11.114,14.556a1.252,1.252,0,0,0,1.768,0L22.568,4.87a.5.5,0,0,0-.281-.849A1.966,1.966,0,0,0,22,4H2a1.966,1.966,0,0,0-.289.021.5.5,0,0,0-.281.849Z"
          />
          <path
            d="M23.888,5.832a.182.182,0,0,0-.2.039l-6.2,6.2a.251.251,0,0,0,0,.354l5.043,5.043a.75.75,0,1,1-1.06,1.061l-5.043-5.043a.25.25,0,0,0-.354,0l-2.129,2.129a2.75,2.75,0,0,1-3.888,0L7.926,13.488a.251.251,0,0,0-.354,0L2.529,18.531a.75.75,0,0,1-1.06-1.061l5.043-5.043a.251.251,0,0,0,0-.354l-6.2-6.2a.18.18,0,0,0-.2-.039A.182.182,0,0,0,0,6V18a2,2,0,0,0,2,2H22a2,2,0,0,0,2-2V6A.181.181,0,0,0,23.888,5.832Z"
          />
        </g>
      </svg>
      Email
    </button>
    <button
      class="css-1gkcy1z"
      type="button"
    >
      Arrow Forward
      <svg
        class="css-g5lpqx"
        data-chakra-custom-icon=""
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`shows spinner and loading text if isLoading and loadingText 1`] = `
<DocumentFragment>
  <button
    aria-disabled="true"
    class="css-1ax3zce"
    data-loading=""
    disabled=""
    type="button"
  >
    <span
      class="css-9pkng7"
    >
      <div
        class="css-1vi2ulp"
      >
        <span
          style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        >
          Loading...
        </span>
      </div>
    </span>
    Submitting
  </button>
</DocumentFragment>
`;

exports[`shows spinner if isLoading 1`] = `
<DocumentFragment>
  <button
    aria-disabled="true"
    class="css-1ax3zce"
    data-loading=""
    disabled=""
    type="button"
  >
    <span
      class="css-18fo44r"
    >
      <div
        class="css-1vi2ulp"
      >
        <span
          style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
        >
          Loading...
        </span>
      </div>
    </span>
    <span
      class="css-q7lffx"
    >
      Email
    </span>
  </button>
</DocumentFragment>
`;
